web_external/User/InviteUserView.js

```diff
@@ -2,28 +2,40 @@ isic.views.InviteUserView = isic.View.extend({
     events: {
         'submit #isic-user-invite-form': function (event) {
             event.preventDefault();
-            // this.$('#isic-user-invite-submit').prop('disabled', true);
+            this.$('#isic-user-invite-submit').prop('disabled', true);
+
+            var data = {
+                login: this.$('#isic-user-invite-new-login').val(),
+                email: this.$('#isic-user-invite-new-email').val(),
+                firstName: this.$('#isic-user-invite-new-firstname').val(),
+                lastName: this.$('#isic-user-invite-new-lastname').val()
+            };
+
+            var validityPeriod = this.$('#isic-user-invite-period').val();
+            if ($.trim(validityPeriod).length > 0) {
+                data.validityPeriod = validityPeriod;
+            }
 
             girder.restRequest({
                 path: 'user/invite',
-                data: {
-                    login: this.$('#isic-user-invite-new-login').val(),
-                    email: this.$('#isic-user-invite-new-email').val(),
-                    firstName: this.$('#isic-user-invite-new-firstname').val(),
-                    lastName: this.$('#isic-user-invite-new-lastname').val()
-                },
+                data: data,
                 type: 'POST',
                 error: null
             })
-                .done(function (resp) {
-                    console.log('success', resp);
-                })
-                .fail(function (resp) {
-                    isic.showAlertDialog({
-                        text: '<h4>Error sending invite</h4><br>' + _.escape(resp.responseJSON.message),
-                        escapedHtml: true
-                    });
+            .done(function (resp) {
+                resp.newUser.email = data.email;
+                girder.events.trigger('g:navigateTo', isic.views.InvitationConfirmationView, {
```

Comment:
FYI, I was initially a little skeptical of the need for a whole separate view to show this, and I was hesitant about directly triggering `g:navigateTo` from inside a view like this, instead of the traditional location in a router.

After further consideration though, I think this is indeed the best way. Associating multiple templates with a single view or packing a large amount of conditional code into a single template both seem less maintainable, and we don't want the confirmation to actually live at another route (which would also make it difficult to pass private parameters to). I'd be fine if we did more of this in the future, where appropriate.